shop/src/main/java/ro/msg/learning/shop/Service/Implementation/ProductService.java

```diff
@@ -0,0 +1,111 @@
+package ro.msg.learning.shop.Service.Implementation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.DTO.ProductAndProductCategoryDto;
+import ro.msg.learning.shop.Domain.Location;
+import ro.msg.learning.shop.Domain.Product;
+import ro.msg.learning.shop.Domain.ProductCategory;
+import ro.msg.learning.shop.Domain.Stock;
+import ro.msg.learning.shop.Exception.*;
+import ro.msg.learning.shop.Mapper.ProductAndProductCategoryTranslatorMapper;
+import ro.msg.learning.shop.Repository.ILocationRepository;
+import ro.msg.learning.shop.Repository.IProductCategoryRepository;
+import ro.msg.learning.shop.Repository.IProductRepository;
+import ro.msg.learning.shop.Repository.IStockRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class ProductService {
+    @Autowired
+    private ProductCategoryService productCategoryService;
+    @Autowired
+    private IProductRepository productRepository;
+    @Autowired
+    private LocationService locationService;
+
+    @Autowired
+    private StockService stockService;
+
+
+
+    public ProductAndProductCategoryDto createProduct (ProductAndProductCategoryDto productAndProductCategoryDto) {
+        //Validate the miss of the product in the product list
+        if (productRepository.existsByAttributes(
+                productAndProductCategoryDto.getProductName(),
+                productAndProductCategoryDto.getProductDescription(),
+                productAndProductCategoryDto.getProductPrice(),
+                productAndProductCategoryDto.getProductWeight(),
+                productAndProductCategoryDto.getProductSupplier(),
+                productAndProductCategoryDto.getProductImageUrl(),
+                productAndProductCategoryDto.getProductCategoryId())) {
+            throw new ProductAlreadyExistingException(productAndProductCategoryDto.getProductName());
+        }
+        //Validate the existence of the product category in the product list
+        ProductCategory productCategory=this.productCategoryService.findProductCategoryById(productAndProductCategoryDto.getProductCategoryId());
+        if(productCategory==null)
+        {
+            throw new ProductCategoryNotFoundException(productAndProductCategoryDto.getProductId());
+        }
+
+        //Validate the existence of the location in the product list
+        Location location=this.locationService.findById(productAndProductCategoryDto.getLocationId());
+
+
+        ProductAndProductCategoryTranslatorMapper mapper=new ProductAndProductCategoryTranslatorMapper();
```

Comment:
We should make the mappers static so we can use them without creating an object like `new .....Mapper()`.